xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/keys/Seed.java

```diff
@@ -192,4 +197,14 @@ public String toString() {
       ", destroyed=" + destroyed +
       '}';
   }
+
+  @Value.Immutable
+  @Wrapped
+  abstract static class Base58EncodedSecret extends Wrapper<String> {
+
+    @Value.Check
+    public void validateBase58EncodedSecret() {
+      assertDoesNotThrow(() -> Base58.decode(this.value()));
+    }
+  }
```

Comment:
This isn't really providing any value as is. We already make sure the `String` you pass in to `fromBase58EncodedSecret` is Base58 encoded on line 125, so now our code has to do 2 Base58 decodes when it should really only need one. Base58 encoding/decoding is an expensive operation, so we don't want to do this unnecessarily. 

I could see this class being useful if we added a derived field containing the decoded `String`, and then using that field on line 125 instead of doing another decode.